# Soru 1:
Weather model

* Load the weather data from Kaggle
* Like in the previous lesson, build a linear regression model where your target variable is the difference between the apparenttemperature and the temperature. As explanatory variables, use humidity and windspeed. Now, estimate your model using OLS. What are the R-squared and adjusted R-squared values? Do you think they are satisfactory? Why?
* Next, include the interaction of humidity and windspeed to the model above and estimate the model using OLS. Now, what is the R-squared of this model? Does this model improve upon the previous one?
* Add visibility as additional explanatory variable to the first model and estimate it. Did R-squared increase? What about adjusted R-squared? Compare the differences put on the table by the interaction term and the visibility in terms of the improvement in the adjusted R-squared. Which one is more useful?
* Choose the best one from the three models above with respect to their AIC and BIC scores. Validate your choice by discussing your justification with your mentor.

In [75]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.linear_model import LinearRegression
from sklearn import linear_model
import statsmodels.api as sm
import warnings
warnings.filterwarnings("ignore")

In [4]:
weather=pd.read_csv("weatherHistory.csv")
weather.head(3)

Formatted Date        Summary Precip Type  Temperature (C)  \
0  2006-04-01 00:00:00.000 +0200  Partly Cloudy        rain         9.472222   
1  2006-04-01 01:00:00.000 +0200  Partly Cloudy        rain         9.355556   
2  2006-04-01 02:00:00.000 +0200  Mostly Cloudy        rain         9.377778   

   Apparent Temperature (C)  Humidity  Wind Speed (km/h)  \
0                  7.388889      0.89            14.1197   
1                  7.227778      0.86            14.2646   
2                  9.377778      0.89             3.9284   

   Wind Bearing (degrees)  Visibility (km)  Loud Cover  Pressure (millibars)  \
0                   251.0          15.8263         0.0               1015.13   
1                   259.0          15.8263         0.0               1015.63   
2                   204.0          14.9569         0.0               1015.94   

                       Daily Summary  
0  Partly cloudy throughout the day.  
1  Partly cloudy throughout the day.  
2  Partly cloudy throughout the day.

In [8]:
weather["diff_temp"]=weather["Temperature (C)"]-weather["Apparent Temperature (C)"]

Y=weather["diff_temp"]
X=weather[["Humidity","Wind Speed (km/h)"]]

lrm=linear_model.LinearRegression()
lrm.fit(X,Y)
print("Coefficient:", lrm.coef_)
print("Bias:", lrm.intercept_)

Coefficient: [3.02918594 0.11929075]
Bias: -2.4381054151877017


In [ ]:
Regresyon formülasyonu:

Y = -2.438 + 3.029*x1 + 0.119*x2

In [13]:
X=sm.add_constant(X)

result=sm.OLS(Y,X).fit()
result.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              diff_temp   R-squared:                       0.288
Model:                            OLS   Adj. R-squared:                  0.288
Method:                 Least Squares   F-statistic:                 1.949e+04
Date:                Sat, 12 Sep 2020   Prob (F-statistic):               0.00
Time:                        20:19:33   Log-Likelihood:            -1.7046e+05
No. Observations:               96453   AIC:                         3.409e+05
Df Residuals:                   96450   BIC:                         3.409e+05
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
=====================================================================================
                        coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------
const                -2.4381      0.021   -115.948      0.000      -2.479      -2.397
Humidity              3.0292      0.024    126.479      0.000       2.982       3.076
Wind Speed (km/h)     0.1193      0.001    176.164      0.000       0.118       0.121
==============================================================================
Omnibus:                     3935.747   Durbin-Watson:                   0.264
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             4613.311
Skew:                           0.478   Prob(JB):                         0.00
Kurtosis:                       3.484   Cond. No.                         88.1
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

R-square değeri ile adjusted R-square değeri aynı olmakla birlikte 0.288'dir. Bu demek oluyorki, tüm katsayılar hedef değişkenin varyansını %28.8 açıklamaktadır, yani hedef değişkeni %28.8 oranında açıkamakta kalan %71.2 oranında hedef değişken hakkında bilgi içermemektedir.Dolayısıyla varyansın büyük olmasını bekleriz, model yeterli değildir.R-square ile adjustedR-square değerin fazla olması hedef değişkeni açıklamada fazla bilgi içerdiğini söyler.

In [27]:
weather["inter_hum_wind"]=weather["Humidity"]*weather["Wind Speed (km/h)"]

Y1=weather["diff_temp"]
X1=weather[["Humidity","Wind Speed (km/h)","inter_hum_wind"]]

lrm1=linear_model.LinearRegression()
lrm1.fit(X1,Y1)

print("Coefficient", lrm1.coef_)
print("Bias", lrm1.intercept_)

Coefficient [-0.17751219 -0.09048213  0.29711946]
Bias -0.0839363100978221


In [ ]:
Humidity ile wind speed ilişkisini özellik olarak modele eklediğimizde çıkan formül:
    
Y1 = -0.084 + -0.177*X11 -0.090*X12 + 0.297*X13

In [28]:
X1=sm.add_constant(X1)

result1=sm.OLS(Y1,X1).fit()
result1.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              diff_temp   R-squared:                       0.341
Model:                            OLS   Adj. R-squared:                  0.341
Method:                 Least Squares   F-statistic:                 1.666e+04
Date:                Sat, 12 Sep 2020   Prob (F-statistic):               0.00
Time:                        20:48:10   Log-Likelihood:            -1.6669e+05
No. Observations:               96453   AIC:                         3.334e+05
Df Residuals:                   96449   BIC:                         3.334e+05
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
=====================================================================================
                        coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------
const                -0.0839      0.033     -2.511      0.012      -0.149      -0.018
Humidity             -0.1775      0.043     -4.133      0.000      -0.262      -0.093
Wind Speed (km/h)    -0.0905      0.002    -36.797      0.000      -0.095      -0.086
inter_hum_wind        0.2971      0.003     88.470      0.000       0.291       0.304
==============================================================================
Omnibus:                     4849.937   Durbin-Watson:                   0.262
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             9295.404
Skew:                           0.378   Prob(JB):                         0.00
Kurtosis:                       4.320   Cond. No.                         193.
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Nem ile rüzgarın ilişkisini modele eklediğimizde R-square ile adjusted R-square değeri %34.1 oranına ulaşarak artmıştır; yani önceden hedef değişkende açıklayamadığı bilgilerin bir kısmını açıklayabilmiştir; dolayısıyla yeni özelliğin eklenmesi modelin tahmin gücünü arttırdığından modelin içinde bu özelliği tutabiliriz.

In [29]:
Y2=weather["diff_temp"]
X2=weather[["Humidity","Wind Speed (km/h)","Visibility (km)"]]
           
lrm2=linear_model.LinearRegression()
lrm2.fit(X2,Y2)
           
print("Coefficient", lrm2.coef_)
print("Bias", lrm2.intercept_)

Coefficient [ 2.60664109  0.11990113 -0.05398318]
Bias -1.5755946860023338


In [ ]:
Visibility değişkeni özellik olarak modele eklendiğinde oluşan formül:
    
Y2 = -1.575 + 2.606*X21 + 0.119*X22 - -0.053*X23

In [30]:
X2=sm.add_constant(X2)

result=sm.OLS(Y2,X2).fit()
result.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              diff_temp   R-squared:                       0.304
Model:                            OLS   Adj. R-squared:                  0.303
Method:                 Least Squares   F-statistic:                 1.401e+04
Date:                Sat, 12 Sep 2020   Prob (F-statistic):               0.00
Time:                        20:50:29   Log-Likelihood:            -1.6938e+05
No. Observations:               96453   AIC:                         3.388e+05
Df Residuals:                   96449   BIC:                         3.388e+05
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
=====================================================================================
                        coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------
const                -1.5756      0.028    -56.605      0.000      -1.630      -1.521
Humidity              2.6066      0.025    102.784      0.000       2.557       2.656
Wind Speed (km/h)     0.1199      0.001    179.014      0.000       0.119       0.121
Visibility (km)      -0.0540      0.001    -46.614      0.000      -0.056      -0.052
==============================================================================
Omnibus:                     3833.895   Durbin-Watson:                   0.279
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             4584.022
Skew:                           0.459   Prob(JB):                         0.00
Kurtosis:                       3.545   Cond. No.                         131.
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Visibility değişkeni özellik olarak modele eklenince; R-square değeri düşerek %30.4, adjusted R-square değeri ise %30.3 olmuştur. Bu demek oluyor ki ilk modele göre visibility özelliği hedef değişkeni açıklamada daha fazla bilgiye sahiptir ancak; humidity ile wind ilişkisini modele eklediğimiz 2. örneğe nazaran daha az R-square değerine sahiptir, dolayısıyla humidity ile wind ilişkisinin modelde yer alması visibilitye göre daha fazla tahmin yapmamızı sağlar.

In [40]:
lrm_R=0.288
lrm1_R=0.341
lrm2_R=0.304

inference=pd.DataFrame(columns=["Hum_wind","Hum_wind_humwind","Hum_wind_visib"])

inference=inference.append({"Hum_wind":lrm_R,
                           "Hum_wind_humwind":lrm1_R,
                           "Hum_wind_visib":lrm2_R}, ignore_index=True)
inference

# 3 durum için de sonuçları yorumlarsak hedef değişken hakkında daha fazla bilgi içeren 2. durumdur yani humidity ile wind
# arasındaki ilişkinin özellik olarak modele eklenmesidir.

#

Hum_wind  Hum_wind_humwind  Hum_wind_visib
0     0.288             0.341           0.304

In [42]:
AIC=340900
BIC=340900
AIC_1=333400
BIC_1=333400
AIC_2=338800
BIC_2=338800

inference_aic_bic=pd.DataFrame(columns=["AIC","AIC_1","AIC_2","BIC","BIC_1","BIC_2"])

inference_aic_bic=inference_aic_bic.append({"AIC":AIC, "AIC_1":AIC_1, "AIC_2":AIC_2, "BIC":BIC, "BIC_1":BIC_1, "BIC_2":BIC_2},
                                          ignore_index=True)

inference_aic_bic

# 3 model için R-square değerlerinde olduğu gibi AIC ve BIC değerleri de en iyi 2. model içindir; çünkü AIC ve BIC değerlerinin
# olabildiğince düşük olması tercih edilir; o yüzden 3 model içinden ben humidity ile wind speed değişken ilişkisinin modele
# özellik olarak eklenmesi modelini kullanmayı tercih ederim.

AIC   AIC_1   AIC_2     BIC   BIC_1   BIC_2
0  340900  333400  338800  340900  333400  338800

# Soru 2:
House Price Model

* Load the houseprices data from Kaggle.
* Run your house prices model again and assess the goodness of fit of your model using F-test, R-squared, adjusted R-squared, AIC and BIC.
* Do you think your model is satisfactory? If so, why?
* In order to improve the goodness of fit of your model, try different model specifications by adding or removing some variables.
* For each model you try, get the goodness of fit metrics and compare your models with each other. Which model is the best and why?

In [45]:
train=pd.read_csv("train.csv")
train.head(3)

Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0   1          60       RL         65.0     8450   Pave   NaN      Reg   
1   2          20       RL         80.0     9600   Pave   NaN      Reg   
2   3          60       RL         68.0    11250   Pave   NaN      IR1   

  LandContour Utilities  ... PoolArea PoolQC Fence MiscFeature MiscVal MoSold  \
0         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      2   
1         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      5   
2         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      9   

  YrSold  SaleType  SaleCondition  SalePrice  
0   2008        WD         Normal     208500  
1   2007        WD         Normal     181500  
2   2008        WD         Normal     223500  

[3 rows x 81 columns]

In [46]:
# Missing values çok fazla olduğu için direkt silebiliriz.

train.drop("LotFrontage",axis=1,inplace=True)
train.drop("Alley",axis=1,inplace=True)
train.drop("Id",axis=1,inplace=True)
train.drop("FireplaceQu",axis=1,inplace=True)
train.drop("PoolQC",axis=1,inplace=True)
train.drop("Fence",axis=1,inplace=True)
train.drop("MiscFeature",axis=1,inplace=True)

In [48]:
train["GarageCond"].fillna(train["GarageCond"].mode()[0],inplace=True)
train["MasVnrArea"].fillna(train["MasVnrArea"].mode()[0],inplace=True)
train["GarageQual"].fillna(train["GarageQual"].mode()[0], inplace=True)
train["GarageFinish"].fillna(train["GarageFinish"].mode()[0], inplace=True)
train["GarageYrBlt"].fillna(train["GarageYrBlt"].mode()[0], inplace=True)
train["GarageType"].fillna(train["GarageType"].mode()[0], inplace=True)
train["Electrical"].fillna(train["Electrical"].mode()[0], inplace=True)
train["BsmtFinType2"].fillna(train["BsmtFinType2"].mode()[0], inplace=True)
train["BsmtFinType1"].fillna(train["BsmtFinType1"].mode()[0], inplace=True)
train["BsmtQual"].fillna(train["BsmtQual"].mode()[0], inplace=True)
train["BsmtCond"].fillna(train["BsmtCond"].mode()[0], inplace=True)
train["BsmtExposure"].fillna(train["BsmtExposure"].mode()[0], inplace=True)
train["MasVnrType"].fillna(train["MasVnrType"].mode()[0], inplace=True)


In [49]:
train["MiscVal"].replace(0,"none",inplace=True)
train["PoolArea"].replace(0,"none",inplace=True)
train["ScreenPorch"].replace(0,"none",inplace=True)
train["3SsnPorch"].replace(0,"none",inplace=True)
train["EnclosedPorch"].replace(0,"none",inplace=True)
train["OpenPorchSF"].replace(0,"none",inplace=True)
train["WoodDeckSF"].replace(0,"none",inplace=True)
train["LowQualFinSF"].replace(0,"none",inplace=True)
train["BsmtFinSF2"].replace(0,"none",inplace=True)

In [50]:
def paved(PavedDrive):
    if PavedDrive=="Y":
        return 3
    elif PavedDrive=="P":
        return 2
    elif PavedDrive=="N":
        return 1

train["PavedDrive_new"]=train["PavedDrive"].apply(paved)

In [51]:
def garagequal(GarageQual):
    if GarageQual=="Ex":
        return 5
    elif GarageQual=="Gd":
        return 4
    elif GarageQual=="TA":
        return 3
    elif GarageQual=="Fa":
        return 2
    elif GarageQual=="Po":
        return 1
    
train["GarageQual_new"]=train["GarageQual"].apply(garagequal)

In [52]:
def garagefinish(GarageFinish):
    if GarageFinish=="Fin":
        return 3
    elif GarageFinish=="RFn":
        return 2
    elif GarageFinish=="Unf":
        return 1

train["GarageFinish_new"]=train["GarageFinish"].apply(garagefinish)

In [53]:
def kitchenqual(KitchenQual):
    if KitchenQual=="Ex":
        return 4
    elif KitchenQual=="Gd":
        return 3
    elif KitchenQual=="TA":
        return 2
    elif KitchenQual=="Fa":
        return 1

train["KitchenQual_new"]=train["KitchenQual"].apply(kitchenqual)

In [54]:
# dummy değişken oluşturma

train["CentralAir_new"]=train["CentralAir"].replace({"Y":1, "N":0})

In [55]:
def extercon(ExterCond):
    if ExterCond=="Ex":
        return 5
    elif ExterCond=="Gd":
        return 4
    elif ExterCond=="TA":
        return 3
    elif ExterCond=="Fa":
        return 2
    elif ExterCond=="Po":
        return 1
    
train["ExterCond_new"]=train["ExterCond"].apply(extercon)

In [77]:
Y3=train["SalePrice"]
X3=train[["GarageArea","GrLivArea","OverallQual","OverallCond","PavedDrive_new","GarageQual_new",
        "GarageFinish_new","KitchenQual_new","CentralAir_new","ExterCond_new"]]

In [78]:
lrm3=linear_model.LinearRegression()
lrm3.fit(X3,Y3)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None,
         normalize=False)

In [73]:
print("Coefficient:", lrm3.coef_)
print("Bias:", lrm3.intercept_)

Coefficient: [   54.88686135    50.56941948 19369.19809456  2146.40285143
  3579.48300454  7034.67649911  9298.5655194  19915.37717733
  5058.61317165 -3717.8078254 ]
Bias: -142700.51819527542


In [74]:
X3=sm.add_constant(X3)
results=sm.OLS(Y3,X3).fit()
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              SalePrice   R-squared:                       0.766
Model:                            OLS   Adj. R-squared:                  0.765
Method:                 Least Squares   F-statistic:                     475.0
Date:                Sat, 12 Sep 2020   Prob (F-statistic):               0.00
Time:                        22:31:13   Log-Likelihood:                -17483.
No. Observations:                1460   AIC:                         3.499e+04
Df Residuals:                    1449   BIC:                         3.505e+04
Df Model:                          10                                         
Covariance Type:            nonrobust                                         
====================================================================================
                       coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------
const            -1.427e+05   1.57e+04     -9.102      0.000   -1.73e+05   -1.12e+05
GarageArea          54.8869      6.148      8.927      0.000      42.827      66.947
GrLivArea           50.5694      2.462     20.540      0.000      45.740      55.399
OverallQual       1.937e+04   1198.475     16.162      0.000     1.7e+04    2.17e+04
OverallCond       2146.4029   1028.059      2.088      0.037     129.759    4163.047
PavedDrive_new    3579.4830   2250.931      1.590      0.112    -835.950    7994.916
GarageQual_new    7034.6765   4289.067      1.640      0.101   -1378.769    1.54e+04
GarageFinish_new  9298.5655   1546.158      6.014      0.000    6265.617    1.23e+04
KitchenQual_new   1.992e+04   2125.099      9.372      0.000    1.57e+04    2.41e+04
CentralAir_new    5058.6132   4579.462      1.105      0.270   -3924.471     1.4e+04
ExterCond_new    -3717.8078   3147.163     -1.181      0.238   -9891.291    2455.675
==============================================================================
Omnibus:                      441.243   Durbin-Watson:                   1.974
Prob(Omnibus):                  0.000   Jarque-Bera (JB):            21252.426
Skew:                           0.615   Prob(JB):                         0.00
Kurtosis:                      21.650   Cond. No.                     2.66e+04
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 2.66e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

Modelin R-square değerinin 0.766 çıkması, özellikler hedef değişkeninin %76.6'sını açıklamaktadır, %23.4'ünü açıklamamıştır.
AIC değeri=34990, BIC=35050 dır. F istatistiği 475 olup p değeri 0 çıkmıştır,bu demektir ki tüm özellikler modelin açıklanmasında yeterlidir model ile indirgenmiş model arasında anlamlı bir fark vardır denir.

In order to improve the goodness of fit of your model, try different model specifications by adding or removing some variables.
For each model you try, get the goodness of fit metrics and compare your models with each other. Which model is the best and why?

In [ ]:
Y = -142700 + GarageArea*5488 + GrLivArea*5056 + OverallQual*19370 + OverallCond*2146 + GarageFinish_new*9298 + 
    KitchenQual_new*19920

In [80]:
Y4=train["SalePrice"]
X4=train[["GarageArea","GrLivArea","OverallQual","OverallCond","GarageFinish_new","KitchenQual_new","TotalBsmtSF"]]

lrm4=linear_model.LinearRegression()
lrm4.fit(X4,Y4)

print("Coef", lrm4.coef_)
print("Bias", lrm4.intercept_)

Coef [   46.11283723    45.86637228 17516.87301295  3029.65520633
  9410.16681426 18548.67768028    28.3294084 ]
Bias -127341.99480394003


In [81]:
X4=sm.add_constant(X4)
result4=sm.OLS(Y4,X4).fit()
result4.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              SalePrice   R-squared:                       0.780
Model:                            OLS   Adj. R-squared:                  0.779
Method:                 Least Squares   F-statistic:                     736.0
Date:                Sat, 12 Sep 2020   Prob (F-statistic):               0.00
Time:                        22:42:20   Log-Likelihood:                -17438.
No. Observations:                1460   AIC:                         3.489e+04
Df Residuals:                    1452   BIC:                         3.493e+04
Df Model:                           7                                         
Covariance Type:            nonrobust                                         
====================================================================================
                       coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------
const            -1.273e+05   7027.514    -18.120      0.000   -1.41e+05   -1.14e+05
GarageArea          46.1128      5.943      7.759      0.000      34.455      57.771
GrLivArea           45.8664      2.389     19.198      0.000      41.180      50.553
OverallQual       1.752e+04   1173.663     14.925      0.000    1.52e+04    1.98e+04
OverallCond       3029.6552    909.408      3.331      0.001    1245.762    4813.549
GarageFinish_new  9410.1668   1482.444      6.348      0.000    6502.207    1.23e+04
KitchenQual_new   1.855e+04   2058.451      9.011      0.000    1.45e+04    2.26e+04
TotalBsmtSF         28.3294      2.804     10.102      0.000      22.829      33.830
==============================================================================
Omnibus:                      527.290   Durbin-Watson:                   1.955
Prob(Omnibus):                  0.000   Jarque-Bera (JB):            70826.372
Skew:                          -0.592   Prob(JB):                         0.00
Kurtosis:                      37.101   Cond. No.                     1.45e+04
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.45e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

İlk modelde R-squared değeri 0.766 iken, TotalBsmtSF değişkenini modele özellik olarak eklediğimizde R-squared değeri artarak 0.780 olmuştur; yani özellikler hedef değişkenin varyansını açıklamada daha fazla bilgiye sahiptir denir. Ayrıca F istatistik değeri de 475'den 736'ya çıkarak, model ile indirgenmiş model arasında anlamlı bir fark oluşmuştur. İlk modelde AIC ve BIC değeri 34990 iken 2.modelde 34890' dır. AIC ve BIC değerinin düşük olması daha önemli olduğu için ikinci modelin performansı daha yüksektir diyebiliriz.

In [97]:
Y5=train["SalePrice"]
X5=train[["GarageArea","GrLivArea","OverallQual","OverallCond","GarageFinish_new","KitchenQual_new","TotalBsmtSF","BsmtUnfSF"]]

lrm5=linear_model.LinearRegression()
lrm5.fit(X4,Y4)

print("Coef", lrm5.coef_)
print("Bias", lrm5.intercept_)

Coef [    0.            46.11283723    45.86637228 17516.87301295
  3029.65520633  9410.16681426 18548.67768028    28.3294084 ]
Bias -127341.9948039348


In [99]:
X5=sm.add_constant(X5)
result5=sm.OLS(Y5,X5).fit()
result5.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              SalePrice   R-squared:                       0.788
Model:                            OLS   Adj. R-squared:                  0.787
Method:                 Least Squares   F-statistic:                     673.3
Date:                Sat, 12 Sep 2020   Prob (F-statistic):               0.00
Time:                        22:59:59   Log-Likelihood:                -17412.
No. Observations:                1460   AIC:                         3.484e+04
Df Residuals:                    1451   BIC:                         3.489e+04
Df Model:                           8                                         
Covariance Type:            nonrobust                                         
====================================================================================
                       coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------
const            -1.247e+05   6916.392    -18.032      0.000   -1.38e+05   -1.11e+05
GarageArea          43.0813      5.856      7.357      0.000      31.594      54.568
GrLivArea           46.3264      2.349     19.722      0.000      41.719      50.934
OverallQual       1.861e+04   1163.327     15.994      0.000    1.63e+04    2.09e+04
OverallCond       2410.7261    897.886      2.685      0.007     649.432    4172.020
GarageFinish_new  8324.4367   1464.715      5.683      0.000    5451.252    1.12e+04
KitchenQual_new   1.855e+04   2023.099      9.169      0.000    1.46e+04    2.25e+04
TotalBsmtSF         34.7669      2.897     12.003      0.000      29.085      40.449
BsmtUnfSF          -17.5830      2.434     -7.224      0.000     -22.357     -12.809
==============================================================================
Omnibus:                      724.411   Durbin-Watson:                   1.949
Prob(Omnibus):                  0.000   Jarque-Bera (JB):           119041.777
Skew:                          -1.222   Prob(JB):                         0.00
Kurtosis:                      47.169   Cond. No.                     1.51e+04
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.51e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

"BsmtUnfSF"  değişkeninin özellik olarak modele eklenmesi R-squared değerini arttırmıi Fistatistiğini arttırmış ve AIC/BIC değerlerini düşürdüğü için modelin performansını arttırmıştır diyebiliriz ancak çok fazla değişiklik olmadığı için modele dahil edilmeyebilir.